tools/integration_tests/gzip/read_gzip_test.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+func genericFileSizeStatTest(t *testing.T, filename string) {
+	mountedFilePath := path.Join(setup.MntDir(), TestBucketPrefixPath, filename)
+	gcsObjectPath := path.Join(setup.TestBucket(), TestBucketPrefixPath, filename)
+	gcsObjectSize, err := helpers.GetGcsObjectSize(gcsObjectPath)
+	if err != nil {
+		t.Fatalf("Failed to get size of gcs object %s: %v\n", gcsObjectPath, err)
+	}
+
+	fi, err := os.Stat(mountedFilePath)
+	if err != nil {
+		t.Fatalf("Failed to get stat info of mounted file %s: %v\n", mountedFilePath, err)
+	}
+
+	if fi.Size() != int64(gcsObjectSize) {
+		t.Fatalf("Size of file mounted through gcsfuse (%s, %d) doesn't match the size of the file on GCS (%s, %d)",
+			mountedFilePath, fi.Size(), gcsObjectPath, gcsObjectSize)
+	}
+}
+
+func genericFullFileReadTest(t *testing.T, filename string) {
+	mountedFilePath := path.Join(setup.MntDir(), TestBucketPrefixPath, filename)
+	data, err := operations.ReadFile(mountedFilePath)
+	if err != nil {
+		t.Errorf("Failed to read mounted file %s: %v", mountedFilePath, err)
+	}
+
+	gcsObjectPath := path.Join(setup.TestBucket(), TestBucketPrefixPath, filename)
+	gcsObjectSize, err := helpers.GetGcsObjectSize(gcsObjectPath)
+	if err != nil {
+		t.Fatalf("Failed to get size of gcs object %s: %v\n", gcsObjectPath, err)
+	}
+	if len(data) != gcsObjectSize {
+		t.Fatalf("Read size for %s (%d bytes) didn't match the size of the GCS object gs://%s (%d bytes)", mountedFilePath, len(data), gcsObjectPath, gcsObjectSize)
+	}
+	if strings.Contains(string(data), helpers.TempFileStrLine) {
+		t.Fatalf("Read data contains '%s' but shouldn't", helpers.TempFileStrLine)
+	}
+}
+
+func genericRangedReadTest(t *testing.T, filename string) {
+	mountedFilePath := path.Join(setup.MntDir(), TestBucketPrefixPath, filename)
+
+	gcsObjectPath := path.Join(setup.TestBucket(), TestBucketPrefixPath, filename)
+	gcsObjectSize, err := helpers.GetGcsObjectSize(gcsObjectPath)
+	if err != nil {
+		t.Fatalf("Failed to get size of gcs object %s: %v\n", gcsObjectPath, err)
+	}
+
+	readSize := gcsObjectSize / 10
+	readOffset := readSize / 10
+	f, err := os.Open(mountedFilePath)
+	if err != nil || f == nil {
+		t.Fatalf("Failed to open local mounted file %s: %v", mountedFilePath, err)
+	}
+
+	buf := make([]byte, readSize)
+	for _, offsetMultiplier := range []int64{1, 3, 5, 7, 9} {
+		n, err := f.ReadAt(buf, offsetMultiplier*int64(readOffset))
+		if err != nil {
+			t.Fatalf("Failed to read mounted file %s: %v", mountedFilePath, err)
+		}
+
+		if readSize != n {
+			t.Fatalf("Read size for %s (%d bytes) didn't match the size of the requested read-size (%d bytes)", mountedFilePath, n, readSize)
+		}
+
+		if strings.Contains(string(buf), helpers.TempFileStrLine) {
+			t.Fatalf("Read buffer contains '%s' but shouldn't", helpers.TempFileStrLine)
+		}
+	}
+}
+
+func TestGzipEncodedTextFileWithNoTransformSize(t *testing.T) {
```

Comment:
Please write comment before each test. like [this](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/operations/copy_dir_test.go#L126)

Reply:
The name of the test function is self-explanatory. The referenced comment lists out files, directories copied by the test. There is no such list for the gzip tests.

Reply:
I've added explainatory comments on the generic*Test functions to explain what they do.